File: ecommerce/src/main/java/com/algolia/instantsearch/examples/ecommerce/views/CategoryOrTypeView.java
package com.algolia.instantsearch.examples.ecommerce.views;

import android.content.Context;
import android.text.Spannable;
import android.util.AttributeSet;

import com.algolia.instantsearch.core.helpers.Highlighter;
import com.algolia.instantsearch.ui.views.AlgoliaHitView;

import org.json.JSONException;
import org.json.JSONObject;


public class CategoryOrTypeView extends NotNullView implements AlgoliaHitView {
    public CategoryOrTypeView(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    @Override
    public void onUpdateView(JSONObject result) {
        try {
            String category = result.getString("category");
            final String attributeToHighlight = !isNull(category) ? "category" : "type";

            final Spannable highlightedAttribute = Highlighter.getDefault()
                    .setInput(result, attributeToHighlight).setStyle(getContext()).render();
            setText(highlightedAttribute);
        } catch (JSONException e) {
            e.printStackTrace();
        }
    }
}

File: querysuggestions/src/main/java/com/algolia/instantsearch/examples/querysuggestions/CategoryOrTypeView.kt
package com.algolia.instantsearch.examples.querysuggestions

import android.content.Context
import android.util.AttributeSet
import com.algolia.instantsearch.helpers.Highlighter
import com.algolia.instantsearch.ui.views.AlgoliaHitView
import org.json.JSONException
import org.json.JSONObject


class CategoryOrTypeView(context: Context, attrs: AttributeSet) : NotNullTextView(context, attrs), AlgoliaHitView {

    override fun onUpdateView(result: JSONObject) {
        try {
            val category = result.optString("category")
            val attributeToHighlight = if (!isNull(category)) "category" else "type"

            text = Highlighter.getDefault().setInput(result, attributeToHighlight).setStyle(context).render()
        } catch (e: JSONException) {
            e.printStackTrace()
        }
    }
}
